### PR TITLE
Allow passing placement constraints through ecsParams yaml file to a task definition

### LINKED ISSUE #721: Add support for placementConstraints within task definition
Per [this comment](https://github.com/aws/amazon-ecs-cli/issues/701#issuecomment-460438898), persisting placement constraints within the task definition (e.g., via `compose create`) would allow them to apply to existing services (by updating the task definition), since service-level placement constraints cannot be editing on existing services.

### PR DESCRIPTION
**Pull request by vadimi**
This PR allows passing placement constraints through ecsParams yaml file to a task definition

Fixes #721

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
- [x] Unit tests passed
- [] Integration tests passed
- [x] Unit tests added for new functionality
- [] Listed manual checks and their outputs in the comments ([example](https://github.com/aws/amazon-ecs-cli/pull/750#issuecomment-472623042))
- [] Link to issue or PR for the integration tests:

**Documentation**
- [] Contacted our doc writer
- [] Updated our README
----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by iamhopaul123** [comment]
Hi @vadimi, can you add some manual tests showing the difference between the previous behavior and the updated behavior? Like #907

File: ecs-cli/modules/cli/compose/project/project.go [hunk 1]
```diff
@@ -21,10 +21,11 @@ import (
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/compose/entity"
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/compose/entity/service"
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/compose/entity/task"
+	"github.com/aws/aws-sdk-go/service/ecs"
 	"github.com/sirupsen/logrus"
 
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
-	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/compose"
+	utils "github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/compose"
```
Comment (iamhopaul123) [inline ecs-cli/modules/cli/compose/project/project.go]: Feel like `composeutils` will be a better name for this. What do you think?

File: ecs-cli/modules/cli/compose/project/project_test.go [hunk 2]
```diff
@@ -21,9 +21,12 @@ import (
 	"testing"
 
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/compose/context"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/compose/entity/service"
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
-	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/compose"
+	cliconfig "github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	utils "github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/compose"
```
Comment (iamhopaul123) [inline ecs-cli/modules/cli/compose/project/project_test.go]: Same here

**Comment by vadimi** [comment]
> Hi @vadimi, can you add some manual tests showing the difference between the previous behavior and the updated behavior? Like #907

@iamhopaul123 yes, absolutely, will add

**Comment by vadimi** [comment]
### Manual testing
**Configuration**

```yaml
# app.yaml
version: '2'
services:
  database:
    image: mysql
    ports:
      - "0:80"
    environment:
      - ENV1=test
```

```yaml
# ecs-params.yaml
version: 1
run_params:
  task_placement:
    constraints:
      - type: memberOf
        expression: "attribute:name == value"
```

**Test before**
```
./ecs-cli compose --file app.yaml --ecs-params ecs-params.yaml --verbose create
DEBU[0000] Parsing the compose yaml...                  
DEBU[0000] Docker Compose version found: 2              
DEBU[0000] Parsing v1/2 project...                      
DEBU[0000] Opening compose files: app.yaml              
DEBU[0000] [0/1] [database]: Adding                     
DEBU[0000] [0/1] [default]: EventType: 32               
DEBU[0000] Parsing the ecs-params yaml...               
DEBU[0000] Parsing the ecs-registry-creds yaml...       
DEBU[0000] Transforming yaml to task definition...      
DEBU[0000] Finding task definition in cache or creating if needed  TaskDefinition="{\n  ContainerDefinitions: [{\n      Command: [],\n      Cpu: 0,\n      DnsSearchDomains: [],\n      DnsServers: [],\n      DockerLabels: {\n\n      },\n      DockerSecurityOptions: [],\n      EntryPoint: [],\n      Environment: [{\n          Name: \"ENV1\",\n          Value: \"test\"\n        }],\n      Essential: true,\n      ExtraHosts: [],\n      Image: \"mysql\",\n      Links: [],\n      LinuxParameters: {\n        Capabilities: {\n\n        },\n        Devices: []\n      },\n      Memory: 512,\n      MountPoints: [],\n      Name: \"database\",\n      PortMappings: [{\n          ContainerPort: 80,\n          HostPort: 0,\n          Protocol: \"tcp\"\n        }],\n      Privileged: false,\n      PseudoTerminal: false,\n      ReadonlyRootFilesystem: false,\n      Ulimits: [],\n      VolumesFrom: []\n    }],\n  Cpu: \"\",\n  ExecutionRoleArn: \"\",\n  Family: \"app\",\n  Memory: \"\",\n  NetworkMode: \"\",\n  TaskRoleArn: \"\",\n  Volumes: []\n}"
DEBU[0000] cache miss                                    taskDef="{\n\n}" taskDefHash=e0741c4dc9d8f6226a9ac21612f90102
INFO[0000] Using ECS task definition                     TaskDefinition="app:9"
```

**Test after**
```
./ecs-cli compose --file app.yaml --ecs-params ecs-params.yaml --verbose create
DEBU[0000] Parsing the compose yaml...
DEBU[0000] Docker Compose version found: 2
DEBU[0000] Parsing v1/2 project...
DEBU[0000] Opening compose files: app.yaml
DEBU[0000] [0/1] [database]: Adding
DEBU[0000] [0/1] [default]: EventType: 32
DEBU[0000] Parsing the ecs-params yaml...
DEBU[0000] Parsing the ecs-registry-creds yaml...
DEBU[0000] Transforming yaml to task definition...
DEBU[0000] Finding task definition in cache or creating if needed  TaskDefinition="{\n  ContainerDefinitions: [{\n      Command: [],\n      Cpu: 0,\n      DnsSearchDomains: [],\n      DnsServers: [],\n      DockerLabels: {\n\n      },\n      DockerSecurityOptions: [],\n      EntryPoint: [],\n      Environment: [{\n          Name: \"ENV1\",\n          Value: \"test\"\n        }],\n      Essential: true,\n      ExtraHosts: [],\n      Image: \"mysql\",\n      Links: [],\n      LinuxParameters: {\n        Capabilities: {\n\n        },\n        Devices: []\n      },\n      Memory: 512,\n      MountPoints: [],\n      Name: \"database\",\n      PortMappings: [{\n          ContainerPort: 80,\n          HostPort: 0,\n          Protocol: \"tcp\"\n        }],\n      Privileged: false,\n      PseudoTerminal: false,\n      ReadonlyRootFilesystem: false,\n      Ulimits: [],\n      VolumesFrom: []\n    }],\n  Cpu: \"\",\n  ExecutionRoleArn: \"\",\n  Family: \"app\",\n  Memory: \"\",\n  NetworkMode: \"\",\n  PlacementConstraints: [{\n      Expression: \"attribute:name == value\",\n      Type: \"memberOf\"\n    }],\n  TaskRoleArn: \"\",\n  Volumes: []\n}"
DEBU[0000] cache miss                                    taskDef="{\n\n}" taskDefHash=e43c348bd8d26f13060a62c194c6eadc
INFO[0000] Using ECS task definition                     TaskDefinition="app:10"
```

With the new changes **Test after** also sends `PlacementConstraints` element with the parameters from ecs-params.yaml to the ecs api.

File: ecs-cli/modules/cli/compose/project/project.go (hunk 1 above)
Comment (vadimi) [reply ecs-cli/modules/cli/compose/project/project.go]: renamed

File: ecs-cli/modules/cli/compose/project/project_test.go (hunk 2 above)
Comment (vadimi) [reply ecs-cli/modules/cli/compose/project/project_test.go]: done

**Comment by allisaurus** [comment]
Hi @vadimi - can you please update this PR to diff with our `dev` branch and rebase on top of that? Thanks!

**Comment by vadimi** [comment]
hi @allisaurus, ah, sorry about that, I should have submitted it against dev in the first place. I rebased it and changed the base branch in the PR.

**Comment by vadimi** [comment]
@iamhopaul123, @allisaurus, @SoManyHs thank you for the review!

**Pull request merged**